Remove memory ordering from LeftRight

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#31026 Remove memory ordering from LeftRight**

This is error prone and probably wrong. Since we don't use LeftRight on the hot path anymore, let's remove this.

Differential Revision: [D18902165](https://our.internmc.facebook.com/intern/diff/D18902165/)

smessmer committed:
Remove memory ordering from LeftRight

This is error prone and probably wrong. Since we don't use LeftRight on the hot path anymore, let's remove this.

Differential Revision: [D18902165](https://our.internmc.facebook.com/intern/diff/D18902165/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove memory ordering from LeftRight"


This is error prone and probably wrong. Since we don't use LeftRight on the hot path anymore, let's remove this.

Differential Revision: [D18902165](https://our.internmc.facebook.com/intern/diff/D18902165/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit 45d97fc:
* **4/4** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=45d97fcdc8e3a8ce3f032ceac102d48ea6a64810).

### 4 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3974515) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (1/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75652589) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75652589))
<details>
<summary>
<code>Dec 17 02:23:38 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected expression</code>
</summary>

```
Dec 17 02:23:38 [ 23%] Generating python/layers/last_n_window_collector.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/arc_cosine_feature_map.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/build_index.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/dropout.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/batch_huber_loss.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/semi_random_features.py 
Dec 17 02:23:38 In file included from /var/lib/jenkins/workspace/c10/util/LeftRight.cpp:1: 
Dec 17 02:23:38 /var/lib/jenkins/workspace/c10/util/LeftRight.h:15:32: error: expected expression 
Dec 17 02:23:38         _counter->fetch_add(1, ); 
Dec 17 02:23:38                                ^ 
Dec 17 02:23:38 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected expression 
Dec 17 02:23:38         _counter->fetch_sub(1, ); 
Dec 17 02:23:38                                ^ 
Dec 17 02:23:38 2 errors generated. 
Dec 17 02:23:38 make[2]: *** [c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o] Error 1 
Dec 17 02:23:38 c10/CMakeFiles/c10.dir/build.make:950: recipe for target 'c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o' failed 
Dec 17 02:23:38 CMakeFiles/Makefile2:2211: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 02:23:38 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 02:23:38 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 02:23:38 [ 23%] Generating python/layers/fc.py 
Dec 17 02:23:38 [ 23%] Generating python/layers/sparse_dropout_with_replacement.py 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3974505) caffe2_onnx_py2_gcc5_ubuntu16_04_build (2/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75653207) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75653207))
<details>
<summary>
<code>Dec 17 02:23:48 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected primary-expression before ')' token</code>
</summary>

```
Dec 17 02:23:48 [ 25%] Generating python/helpers/dropout.py 
Dec 17 02:23:48 [ 25%] Generating python/helpers/fc.py 
Dec 17 02:23:48 [ 25%] Generating python/helpers/arg_scope.py 
Dec 17 02:23:48 [ 25%] Generating python/helpers/nonlinearity.py 
Dec 17 02:23:48 In file included from /var/lib/jenkins/workspace/c10/util/LeftRight.cpp:1:0: 
Dec 17 02:23:48 /var/lib/jenkins/workspace/c10/util/LeftRight.h: In constructor 'c10::detail::IncrementRAII::IncrementRAII(std::atomic<int>*)': 
Dec 17 02:23:48 /var/lib/jenkins/workspace/c10/util/LeftRight.h:15:32: error: expected primary-expression before ')' token 
Dec 17 02:23:48          _counter->fetch_add(1, ); 
Dec 17 02:23:48                                 ^ 
Dec 17 02:23:48 /var/lib/jenkins/workspace/c10/util/LeftRight.h: In destructor 'c10::detail::IncrementRAII::~IncrementRAII()': 
Dec 17 02:23:48 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected primary-expression before ')' token 
Dec 17 02:23:48          _counter->fetch_sub(1, ); 
Dec 17 02:23:48                                 ^ 
Dec 17 02:23:48 c10/CMakeFiles/c10.dir/build.make:950: recipe for target 'c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o' failed 
Dec 17 02:23:48 make[2]: *** [c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o] Error 1 
Dec 17 02:23:48 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 02:23:48 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 02:23:48 CMakeFiles/Makefile2:2211: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 02:23:48 [ 25%] Generating python/helpers/db_input.py 
Dec 17 02:23:48 [ 25%] Generating python/helpers/elementwise_linear.py 
Dec 17 02:23:48 [ 25%] Generating python/cached_reader.py 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3974502) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (3/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75653942) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75653942))
<details>
<summary>
<code>Dec 17 02:21:58 /pytorch/c10/util/LeftRight.h:19:32: error: expected primary-expression before ‘)’ token</code>
</summary>

```
Dec 17 02:21:56 [ 28%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/duration.pb.cc.o 
Dec 17 02:21:56 [ 28%] Building CXX object c10/CMakeFiles/c10.dir/util/Half.cpp.o 
Dec 17 02:21:57 [ 28%] Building CXX object c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o 
Dec 17 02:21:57 [ 28%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/dynamic_message.cc.o 
Dec 17 02:21:58 In file included from /pytorch/c10/util/LeftRight.cpp:1:0: 
Dec 17 02:21:58 /pytorch/c10/util/LeftRight.h: In constructor ‘c10::detail::IncrementRAII::IncrementRAII(std::atomic<int>*)’: 
Dec 17 02:21:58 /pytorch/c10/util/LeftRight.h:15:32: error: expected primary-expression before ‘)’ token 
Dec 17 02:21:58          _counter->fetch_add(1, ); 
Dec 17 02:21:58                                 ^ 
Dec 17 02:21:58 /pytorch/c10/util/LeftRight.h: In destructor ‘c10::detail::IncrementRAII::~IncrementRAII()’: 
Dec 17 02:21:58 /pytorch/c10/util/LeftRight.h:19:32: error: expected primary-expression before ‘)’ token 
Dec 17 02:21:58          _counter->fetch_sub(1, ); 
Dec 17 02:21:58                                 ^ 
Dec 17 02:21:58 gmake[2]: *** [c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o] Error 1 
Dec 17 02:21:58 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 02:21:58 gmake[1]: *** Waiting for unfinished jobs.... 
Dec 17 02:21:58 [ 28%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_generic.dir/src/FbgemmSpMM.cc.o 
Dec 17 02:21:58 [ 28%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/empty.pb.cc.o 
Dec 17 02:21:59 [ 28%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/extension_set_heavy.cc.o 
Dec 17 02:21:59 [ 28%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/cpu_primitive.cpp.o 
Dec 17 02:22:00 [ 28%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/cpu_reducer.cpp.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3974512) pytorch_linux_xenial_py3_clang5_mobile_build (4/4)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75653984) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75653984))
<details>
<summary>
<code>Dec 17 02:24:33 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected primary-expression before ')' token</code>
</summary>

```
Dec 17 02:24:33 [ 21%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/x86/cache/descriptor.c.o 
Dec 17 02:24:33 [ 21%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/x86/cache/deterministic.c.o 
Dec 17 02:24:33 [ 21%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/x86/linux/init.c.o 
Dec 17 02:24:33 [ 21%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/x86/linux/cpuinfo.c.o 
Dec 17 02:24:33 In file included from /var/lib/jenkins/workspace/c10/util/LeftRight.cpp:1:0: 
Dec 17 02:24:33 /var/lib/jenkins/workspace/c10/util/LeftRight.h: In constructor 'c10::detail::IncrementRAII::IncrementRAII(std::atomic<int>*)': 
Dec 17 02:24:33 /var/lib/jenkins/workspace/c10/util/LeftRight.h:15:32: error: expected primary-expression before ')' token 
Dec 17 02:24:33          _counter->fetch_add(1, ); 
Dec 17 02:24:33                                 ^ 
Dec 17 02:24:33 /var/lib/jenkins/workspace/c10/util/LeftRight.h: In destructor 'c10::detail::IncrementRAII::~IncrementRAII()': 
Dec 17 02:24:33 /var/lib/jenkins/workspace/c10/util/LeftRight.h:19:32: error: expected primary-expression before ')' token 
Dec 17 02:24:33          _counter->fetch_sub(1, ); 
Dec 17 02:24:33                                 ^ 
Dec 17 02:24:33 make[2]: *** [c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o] Error 1 
Dec 17 02:24:33 c10/CMakeFiles/c10.dir/build.make:950: recipe for target 'c10/CMakeFiles/c10.dir/util/LeftRight.cpp.o' failed 
Dec 17 02:24:33 CMakeFiles/Makefile2:750: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 02:24:33 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 02:24:33 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 02:24:33 [ 22%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/linux/smallfile.c.o 
Dec 17 02:24:33 [ 22%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/linux/multiline.c.o 
Dec 17 02:24:33 [ 22%] Building C object confu-deps/cpuinfo/CMakeFiles/cpuinfo_internals.dir/src/linux/current.c.o 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

albanD commented on c10/util/LeftRight.h:
```diff
@@ -12,11 +12,11 @@ namespace detail {
 public:
     explicit IncrementRAII(std::atomic<int32_t> *counter): _counter(counter) {
-        _counter->fetch_add(1, std::memory_order_acquire);
+        _counter->fetch_add(1, );
```
nit: Why keep the `, ` ?

smessmer replied:
fixed

smessmer committed:
Update on "Remove memory ordering from LeftRight"


This is error prone and probably wrong. Since we don't use LeftRight on the hot path anymore, let's remove this.

Differential Revision: [D18902165](https://our.internmc.facebook.com/intern/diff/D18902165/)

[ghstack-poisoned]

albanD commented:
Are the ci errors relevant? Or they just need a rebase on top of master?

smessmer committed:
Update on "Remove memory ordering from LeftRight"


This is error prone and probably wrong. Since we don't use LeftRight on the hot path anymore, let's remove this.

Differential Revision: [D18902165](https://our.internmc.facebook.com/intern/diff/D18902165/)

[ghstack-poisoned]

smessmer commented:
@albanD CI errors are unrelated
